### DISCUSSION TITLE
Is there such a thing as after attach callback (on associations)?

### DISCUSSION THREAD

**Discussion by glaucocustodio** [avo-hq/avo#2041] [Help]
Given:
```ruby
class UserResource < Avo::BaseResource
  field :organisations,
    as: :has_many,
    through: :memberships
end
```
```ruby
class OrganisationResource < Avo::BaseResource
  field :users,
    as: :has_many,
    through: :memberships
end
```
Does Avo provide a callback that gets triggered once an organisation has been attached to a user (or when a user has been attached to an organisation)?

I think of something like:
```ruby
class Avo::UsersController < Avo::ResourcesController
  def after_organisations_attached
    # send an email
  end
end
```

We could have an `after_commit :send_email, on: :create` on the `Membership` model but in this case we don't know whether or not the creation is happening through Avo.

**Comment by adrianthedev** [comment]
Not at the moment, but we'd support a PR to have this feature added. Something similar can be done with detaching, right?

**Reply by glaucocustodio** [reply]
is there a way to know when a model record is being create through Avo?

**Reply by adrianthedev** [reply]
Do the `before|after_create` hooks not help?

**Comment by adrianthedev** [comment]
BTW, does Rails have anything like that? I feel that it should be a Rails feature rather than Avo feature.
